### ISSUE TITLE
Docker does not work within the homestead.

### ISSUE THREAD

**Issue by alex-oliveira** [laravel/homestead#1428]
## Versions
  - Vagrant 2.2.9
  - Virtualbox: 6.1.8r137981
  - Homestead: v10.8.0

## Host operating system
 - OS Name: Microsoft Windows 10 Pro
 - Version: 10.0.18363 Build 18363

## Homestead.yaml
```yaml
---
ip: "192.168.10.10"
memory: 4056
cpus: 2
provider: virtualbox

authorize: ~/.ssh/id_rsa.pub

keys:
    - ~/.ssh/id_rsa

folders:
    - map: D:/Workspace/www/homestead
      to: /home/vagrant/www/homestead

sites:
    - map: homestead.loc
      to: /home/vagrant/www/homestead/public

databases:
    - homestead

features:
    - mariadb: false
    - ohmyzsh: false
    - webdriver: false
    - docker: true
```

## Vagrant destroy & up output
https://gist.github.com/alex-oliveira/8876964babdb099139cfb78e3d2b31b2

## Expected behavior
I think I should be able to use the docker normally inside the homestead.

## Actual behavior
Apparently, there are several errors related to the "hash" comparison. This prevents perfect installation.

## Steps to reproduce
1. Just add "- docker: true" to the file and try to run "vagrant up".
2. If you remove "- docker: true", everything works fine.

**Comment by alex-oliveira** [comment]
The error messages are not always the same.
In another attempt I obtained:

homestead: # Executing docker install script, commit: 26ff363bcf3b3f5a00498ac43694bf1c7d9ce16c
    homestead: + sh -c 'apt-get update -qq >/dev/null'
    homestead: E: Problem parsing dependency 21
    homestead: E: Error occurred while processing lib32gcc-5-dev-ppc64-cross (NewVersion2)
    homestead: E: Problem with MergeList /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_bionic_universe_binary-amd64_Packages
    homestead: E: The package lists or status file could not be parsed or opened.
    homestead: usermod: group 'docker' does not exist
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

**Comment by svpernova09** [comment]
Unable to reproduce :/

https://gist.github.com/svpernova09/d2f81e5ca7f71dec69512db75a377aa9

**Comment by Ecareyu** [comment]
I also encountered this problem

**Comment by svpernova09** [comment]
@Ecareyu Unless you're proving help on how to reproduce the issue your issue doesn't help anyone :/

**Comment by Ecareyu** [comment]
@alex-oliveira I reinstalled vagrant homestead and re-downloaded the box. After trying many times, I successfully installed docker through the feature switch. However, I still don’t know why I reported this error.

**Comment by Ecareyu** [comment]
@alex-oliveira I reinstalled vagrant virtualbox and re-downloaded the box. After trying many times, I successfully installed docker through the feature switch. However, I still don’t know why I reported this error.

**Comment by alex-oliveira** [comment]
I managed to make it work, but I messed with so much that I can't say for sure what the solution was.

After suffering a lot trying to configure the homestead, I noticed that other virtual machines were also having problems.

I also couldn't create new machines any more.

After some research, I found that installing the docker on the host machine can cause problems for virtual machines.

I also read random reports of people using WSL, having problems with virtual machines.

With that in mind, I decided to uninstall the docker, WSL and other junk from the host machine.

After doing this, the virtual machines started working again and the homestead docker managed to run without errors.
